Fix default environments in guide (#46849)
rails/rails: guides/source/debugging_rails_applications.md

This is useful when you want to log under development or staging without flooding your production log with unnecessary information.

> TIP: The default Rails log level is `debug` for `development` and `test` enviroments. And `info` for `production` environment.

Reviewer: ```suggestion
TIP: The default Rails log level is `debug` for the `development` and `test` environments. And `info` for the `production` environment.
```

> TIP: The default Rails log level is `debug` for `development` and `test` environments. And `info` for the `production` environment.

Reviewer 2: To be more precise, the default is always `:debug`, but it is overridden [in `config/environments/production.rb`](https://github.com/rails/rails/blob/663e8e1ccd049dc005924f7033ae407d2d25ba8a/railties/lib/rails/generators/rails/app/templates/config/environments/production.rb.tt#L61).  So what about something like:

```suggestion
TIP: The default Rails log level is `:debug`. However, it is set to `:info` for the `production` environment in the default generated `config/environments/production.rb`.
```
